pkg/plugins/backendplugin/manager/manager.go

```diff
@@ -244,14 +285,15 @@ func (m *manager) stop(ctx context.Context) {
 
 // CollectMetrics collects metrics from a registered backend plugin.
 func (m *manager) CollectMetrics(ctx context.Context, pluginID string) (*backend.CollectMetricsResult, error) {
-	m.pluginsMu.RLock()
-	p, registered := m.plugins[pluginID]
-	m.pluginsMu.RUnlock()
-
+	p, registered := m.Get(pluginID)
 	if !registered {
 		return nil, backendplugin.ErrPluginNotRegistered
 	}
 
+	if providerID, exists := m.GetProviderForPlugin(pluginID); exists {
```

Comment:
Current hack for providing the actual provider and allow a provider to proxy/bundle/support serving multiple providers. This might be a property on `backend.PluginContext` eventually and/or included in context/gRPC metadata to allow proxying provider gRPC requests without unpacking/decode the full gRPC payload.